src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -103,11 +96,45 @@
 #define ICM426XX_INT_TPULSE_DURATION_100            (0 << ICM426XX_INT_TPULSE_DURATION_BIT)
 #define ICM426XX_INT_TPULSE_DURATION_8              (1 << ICM426XX_INT_TPULSE_DURATION_BIT)
 
-
 #define ICM426XX_RA_INT_SOURCE0                     0x65
 #define ICM426XX_UI_DRDY_INT1_EN_DISABLED           (0 << 3)
 #define ICM426XX_UI_DRDY_INT1_EN_ENABLED            (1 << 3)
 
+typedef struct odrConfig_s {
+    uint8_t odr;            // ODR: Output data rate (gyro sample rate in kHz)
+    uint8_t odr_idx;        // See GYRO_ODR in datasheet
+} odrConfig_t;
+
+typedef struct aafConfig_s {
+    uint8_t delt;
+    uint16_t deltSqr;
+    uint8_t bitshift;
+} aafConfig_t;
```

Comment:
Something like this:

```suggestion
#define AAF(id, delt, deltSqr, bitshift) [(id)] = {{(delt), (deltSqr) & 0xFF, (deltSqr) >> 8) | (bitshift) << 4}} \
/**/                                                                    
struct aafConfig_s {
   uint8_t aaf[3];
} aafLUT[] = {
AAF(AAF_CONFIG_258HZ,  6,   36, 10),  //  258 Hz
AAF(AAF_CONFIG_536HZ, 12,  144,  8), //  536 Hz
AAF(AAF_CONFIG_997HZ, 21,  440,  6), //  997 Hz
AAF(AAF_CONFIG_1962HZ, 37, 1376,  4), // 1962 Hz
};
#undef AAF

uint8_t gyro_aaf_map = {
 [GYRO_HARDWARE_LPF_NORMAL] = AAF_CONFIG_258HZ,
 [GYRO_HARDWARE_LPF_OPTION_1] = AAF_CONFIG_536HZ,
 [GYRO_HARDWARE_LPF_OPTION_2] = AAF_CONFIG_997HZ,
#ifdef USE_GYRO_DLPF_EXPERIMENTAL
[GYRO_HARDWARE_LPF_EXPERIMENTAL] = AAF_CONFIG_1962HZ,
#endif
};
```

```
unsigned gyro = (gyroConfig()->gyro_hardware_lpf < ARRAYLEN(gyro_aaf_map)) ? gyro_aaf_map[gyroConfig()->gyro_hardware_lpf) : 0;
// AAF_CONFIG_258HZ == 0 is used as default value

for (unsigned i = 0; i < 3; i++)
   spiWriteReg(dev, ICM426XX_RA_GYRO_CONFIG_STATIC3 + i, aafLUT[gyro].aaf[i]);

for (unsigned i = 0; i < 3; i++)
   spiWriteReg(dev, ICM426XX_RA_ACCEL_CONFIG_STATIC2 + i, aafLUT[AAF_CONFIG_258HZ].aaf[i] << (i==0 ? 1 : 0));
```

Reply:
I'm totally against this as this overcomplicates the simplest of things. My goal was to get readable code. Let's just use a neat and clearly laid out lookup table that is 1 to 1 equal to the documentation.

Leaving it as is also makes it very transparent to see how the settings get written into the registers.

Reply:
I can see your point. For me, shorter code is easier to read ;-)

TODO:
- USE_GYRO_DLPF_EXPERIMENTAL
- posibly reorder aafConfig_s to get better packing

Reply:
Ok, so I tried to get better packing but the compiler seems to make a good job already. I used `__attribute__((__packed__))`, put `deltSqr` first and even as the last attribute. But the outcome is still the same, so I left it as is.

Reply:
The difference is small, so linker will probably hide the difference.
Struct with uint16_t must be 2 bytes aligned and it's size must be multiple of 2.
so actual layout is `{uint8_t delt; uint8_t padding_1;   uint16_t deltSqr; uint8_t bitshift; uint8_t padding_2; }`
Using `__attribute__((__packed__))` (IMO there is `PACKED` define in BF) is overkill here. But putting `bitshift` first may be worth it (but it's at most 16 bytes FLASH/RAM)